### PR TITLE
feat: simplify `init` and `link` flows

### PR DESCRIPTION
**Pull request by eduardoboucas**
#### Summary

This PR contains a few changes to streamline the `init` and `dev` commands:

- Remove message in `link` command pointing to the location of the `state.json` file: https://github.com/netlify/cli/commit/0322ed5949d3807d5c2a68e7e298b5075f458703
- Automatically install recommended plugins in `init` command: https://github.com/netlify/cli/commit/8b4d91c9383de8b906bb2cbbe9ad37e1e94c1139
- Stop prompting users for the functions directory and use default instead: https://github.com/netlify/cli/commit/efc45b5f32582826cdf99bf463b2eb7f74017fcf
- Remove location of `state.json` in `link` confirmation message: https://github.com/netlify/cli/commit/62c566cf5bb8e33a63c7b8e055753ef727c55b24
- Add blank line around prompts: https://github.com/netlify/cli/commit/8d317004d963fea7d195cc06b4d873b91575d175
- Add "Configuring <framework> runtime..." message: https://github.com/netlify/cli/commit/8a3e1828a27525a79bee69c3bde7c22ad166b6a9
- Stop showing `undefined` when default site name is chosen: https://github.com/netlify/cli/commit/c3b1a69e9cdfc3ec82be900a6b9e75d22e197d5f

### CODE REVIEW

File: src/utils/init/utils.js [hunk 1]
```diff
@@ -154,25 +106,24 @@ const getBuildSettings = async ({ config, env, repositoryRoot, siteRoot }) => {
       frameworkBuildDir,
       frameworkPlugins,
     })
-  const { baseDir, buildCmd, buildDir, functionsDir, installSinglePlugin, plugins } = await inquirer.prompt(
-    await getPromptInputs({
+
+  if (recommendedPlugins.length !== 0) {
+    log(`Configuring ${formatTitle(frameworkName)} runtime...`)
```
Comment (ascorbic) [inline src/utils/init/utils.js:111]:
```suggestion
    log(`Configuring ${formatTitle(frameworkName)}...`)
```

It has "runtime" in the plugin name already

Comment (ascorbic) [reply src/utils/init/utils.js:111]: Oh wait, you're using the framework name not plugin name. Carry on!

File: tests/integration/420.command.init.test.js [hunk 2]
```diff
@@ -491,7 +473,7 @@ test('netlify init existing Next.js site with existing plugins', async () => {
           provider: 'manual',
           repo_branch: 'main',
           repo_path: 'git@github.com:owner/repo.git',
-          functions_dir: functions,
+          functions_dir: 'netlify/functions',
```
Comment (ascorbic) [inline tests/integration/420.command.init.test.js:476]: Is `defaultFunctionsDirectory` in scope here?

File: src/commands/sites/sites-create.js [hunk 3]
```diff
@@ -39,16 +39,14 @@ const getSiteNameInput = async (name, user, api) => {
...
-      `Choose a unique site name (e.g. ${siteSuggestion}.netlify.app) or leave it blank for a random name. You can update the site name later.`,
-    )
     const { name: nameInput } = await inquirer.prompt([
       {
         type: 'input',
         name: 'name',
-        message: 'Site name (optional):',
-        filter: (val) => (val === '' ? undefined : val),
-        validate: (input) => /^[a-zA-Z\d-]+$/.test(input) || 'Only alphanumeric characters and hyphens are allowed',
+        message: 'Site name (you can change it later):',
```
Comment (whitep4nth3r) [inline src/commands/sites/sites-create.js:46]: Can you change the name later on the CLI or JUST in app.netlify? Might be worth calling that out?

**Review by ascorbic — DISMISSED** [review]
Love it

File: src/utils/init/utils.js (hunk 1 above)
Comment (eduardoboucas) [reply src/utils/init/utils.js:111]: Yeah. I could use the plugin name, but theoretically you could have many plugins (even though I suspect in reality we never do). I'll use the framework for now, and we can revisit later.

File: src/commands/sites/sites-create.js (hunk 3 above)
Comment (eduardoboucas) [reply src/commands/sites/sites-create.js:46]: You can update in the CLI too, using the `ntl api` command.

File: tests/integration/420.command.init.test.js (hunk 2 above)
Comment (eduardoboucas) [reply tests/integration/420.command.init.test.js]: Oops! It is. I missed this one. Updated in https://github.com/netlify/cli/pull/4979/commits/a046cc59cd5ac6eeb7f42d84945ca012fba52911.

**Review by ascorbic — APPROVED** [review]

**Pull request merged**